src/librustc/session/config.rs

```diff
@@ -2304,10 +2310,26 @@ pub fn build_session_options_and_crate_config(
             );
         };
 
+
         externs
             .entry(name.to_owned())
-            .or_default()
```

Comment:
If `ExternEntry` derives `Default`, then code here and in rustdoc will became simpler (no `or_insert_with` required).